### ISSUE TITLE
[BUG] str comprehension for event logging in protocols when using fancy parameters

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#403]
**Describe the bug**
```
Traceback (most recent call last):
  File "c:\users\bergamot\code\python-microscopy\PYME\ui\mytimer.py", line 42, in Notify
    a()
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\ActionManager.py", line 140, in Tick
    self.isLastTaskDone = fcn(**args)
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\SpoolController.py", line 420, in StartSpooling
    self.spooler.StartSpool()
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\HTTPSpooler.py", line 229, in StartSpool
    sp.Spooler.StartSpool(self)
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\Spooler.py", line 133, in StartSpool
    self.protocol.Init(self)
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\protocol.py", line 121, in Init
    self.OnFrame(-1) #do everything which needs to be done before acquisition starts
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\protocol.py", line 131, in OnFrame
    eventLog.logEvent('ProtocolTask', '%d, %s, ' % (frameNum, t.what.__name__) + ', '.join(['%s' % p for p in t.params]))
  File "c:\users\bergamot\code\python-microscopy\PYME\Acquire\protocol.py", line 131, in <listcomp>
    eventLog.logEvent('ProtocolTask', '%d, %s, ' % (frameNum, t.what.__name__) + ', '.join(['%s' % p for p in t.params]))
TypeError: not all arguments converted during string formatting
```

getting thrown on the following task list
```
taskList = [
    T(-1, scope.l405.SetPower, 1),
    T(-1, scanner.set_state, [1], (304, 304), 0.004),
    T(-1, scope.focus_lock.EnableLock),  # should already be enabled, but just in case
    T(-1, scope.l405.TurnOn),
    T(-1, scanner.genCoords),
    T(0, scanner.start),
    T(maxint, scope.turnAllLasersOff),
    T(maxint, scanner.return_state)
]
```

We take *args on the 'what' parameter, so `scanner.set_state` executes fine, but in the log str formatting we don't handle non-scalar parameters:
```
', '.join(['%s' % p for p in ((304, 304), 3)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
TypeError: not all arguments converted during string formatting
```

**To Reproduce**
1. write a protocol with a tuple/list as a parameter
2. run it

**Expected behavior**
convert tuples to str
```
>>> '%s' % (304, 304)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: not all arguments converted during string formatting
>>> '%s' % str((304, 304))
'(304, 304)'
```

**Screenshots**

**Platform (please complete the following information - or copy-paste from error dialog if shown):**

 - Python version: 3

**Comment by barentine** [comment]
my repo is a bit of a mess here in lab but I'll sub a pr when I'm home

**Comment by barentine** [comment]
fixed in #407
